packages/shared/utils/scriptureMarkers/getScripturalMarkerActions.ts

```diff
@@ -0,0 +1,315 @@
+import { UsjNode } from "../../converters/usj/core/usj";
+import { usjNodeToSerializedLexical } from "../../converters/usj";
+import { ScriptureElementNode } from "../../nodes/scripture/generic/ScriptureElementNode";
+
+export const markerActions: {
+  [marker: string]: {
+    label?: string;
+    action?: (currentEditor: {
+      reference: { book: string; chapter: number; verse: number };
+      editor: LexicalEditor;
+    }) => SerializedLexicalNode | string;
+  };
+} = {
+  no: {
+    label: "format_clear",
+  },
+  it: {
+    label: "format_italic",
+  },
+  bd: {
+    label: "format_bold",
+  },
+  sc: {
+    label: "uppercase",
+  },
+  sup: {
+    label: "superscript",
+  },
+  v: {
+    action: (currentEditor) => {
+      const { verse } = currentEditor.reference;
+      return String.raw`\v ${verse + 1} `;
+    },
+  },
+  c: {
+    action: (currentEditor) => {
+      const { chapter } = currentEditor.reference;
+      return String.raw`\p \c ${chapter + 1} `;
+    },
+  },
+  f: {
+    action: (currentEditor) => {
+      const { chapter, verse } = currentEditor.reference;
+      return String.raw`\f + \fr ${chapter}.${verse}: \ft ${CURSOR_PLACEHOLDER_CHAR} \f*`;
+    },
+  },
+  x: {
+    action: (currentEditor) => {
+      const { chapter, verse } = currentEditor.reference;
+      return String.raw`\x + \xo ${chapter}.${verse}: \xt ${CURSOR_PLACEHOLDER_CHAR} \x*`;
+    },
+  },
+};
+
+function createEmptyUsj(
+  marker: string,
+  markerData: ScripturalMarker,
+  reference: { book: string; chapter: number; verse: number },
+): UsjNode | undefined {
+  const element = markerData.element;
+
+  if (!element) {
+    return undefined;
+  }
+
+  // Helper function to get required attributes based on marker type
+  const getRequiredAttributes = () => {
+    const bookCode = reference.book.toUpperCase();
+    const bcId = `${bookCode}.${reference.chapter}`;
+    switch (marker) {
+      case "v":
+        return {
+          number: String(reference.verse + 1),
+          sid: `${bcId}.${Number(reference.verse) + 1}`,
+        };
+      case "c":
+        return {
+          number: String(reference.chapter + 1),
+          sid: `${bookCode}.${Number(reference.chapter) + 1}`,
+        };
+      case "book":
+        return { code: bookCode };
+      case "f":
+        return {
+          caller: "+",
+          content: [
+            { type: "char", marker: "fr", content: [`${reference.chapter}.${reference.verse}`] },
+            {
+              type: "char",
+              marker: "ft",
+              content: [" "],
+            },
+          ],
+        };
+      case "x":
+        return {
+          caller: "+",
+          content: [
+            { type: "char", marker: "xo", content: [`${reference.chapter}.${reference.verse}`] },
+            { type: "char", marker: "xt", content: [" "] },
+          ],
+        };
+      default:
+        return {};
+    }
+  };
+
+  let nodeType: UsjNode["type"] | undefined = undefined;
```

Comment:
NIT: assigning to `undefined` isn't needed since declaring it makes it undefined.